umap/static/umap/js/modules/sync/updaters.js

```diff
@@ -53,18 +53,22 @@ export class MapUpdater extends BaseUpdater {
 }
 
 export class DataLayerUpdater extends BaseUpdater {
-  upsert({ value }) {
-    // Inserts does not happen (we use multiple updates instead).
-    this.map.createDataLayer(value, false)
+  upsert({ value, metadata }) {
+    console.log('upsert', value, metadata)
+    // Upsert only happens when a new datalayer is created.
+    this.map.createDataLayer(value, false, metadata.future_uuid)
```

Comment:
Here I would expect the uuid to be in the value, not separate.

Reply:
It *could* also be in the value, but since it's not yet saved on the server I believe it's better to keep it this way. After all, metadata is that: a way to identify the data. Once it will be saved, then it will be part of the data (and also be sent in the metadata)